db_stress_tool/no_batched_ops_stress.cc

```diff
@@ -1309,7 +1309,9 @@ class NonBatchedOpsStressTest : public StressTest {
       }
     }
 
-    pending_expected_value.Commit();
+    if (!FLAGS_use_optimistic_txn || s.ok()) {
```

Comment:
If transaction commit fails, why won't we call the SafeTerminate() below?

Reply:
Hm.. Not sure if I understand your question. We will call `SafeTerminate()` when the transaction fails.

The theory we have for the test failure is that, if the db stress crashes _after_ `pending_expected_value.Commit()` but _before_ `thread->shared->SafeTerminate();, we may have expected value committed while actual value DB was not.

The change I am trying to make here is not to commit pending expected value if the failed transaction is optimistic transaction.

Reply:
Oh I see. I did not think of it crashing in between. Would it work by just moving pending_expected_value.Commit() after !s.ok() check?

Reply:
Let me give it a shot. I first wanted to limit the impact to be optimistic txn db only, but I think that's the right way to fix this.